### DISCUSSION TITLE
[6팀 ark] 동행 게시글 생성, 부하 증가 시나리오

### DISCUSSION THREAD

**Discussion by 49EHyeon42** [100-hours-a-week/discussions#87] [시나리오]
### 시나리오

6팀의 커넥트립 홍보가 대성공을 거두며, 사용자 수가 기하급수적으로 증가하였다. 커넥트립에서 동행 커뮤니티가 생성될 때마다, 게시글과 그룹 채팅방이 함께 생성되며, 웹소켓을 통해 실시간 채팅이 가능하도록 구현되었다. 이러한 구조로 인해 서버 부하가 급격히 증가할 것으로 예상된다.

### 순서

1. 사용자가 1분에 1,000명씩 점진적으로 늘어난다.
2. 총 5분 동안 5,000명이 동행 게시글을 작성한다.
3. 사용자는 `/accompany`를 시작으로 `/accompany/create` 엔드포인트를 통해 동행 게시글을 작성한다.
4. 게시글 생성과 동시에 그룹 방 그리고 웹소켓을 활용한 채팅 관련 테이블에 행이 추가된다.
5. 최종적으로 성능에서 병목이 있진 않은지 관찰하기 위해 다음을 확인한다.
   1.  HTTP Response TIme
   2. HTTP Response Status Code
   3. HTTP Request P95, P99
   4. 서버측 metric datas (CPU, RAM, etc.)
   5. 데이터베이스 부하 확인

### Database Query

사용자가 동행 게시글을 생성하면 아래의 쿼리가 발생한다.

```
# 사용자 조회
Hibernate: 
    select
        m1_0.id,
        생략
    from
        member m1_0 
    where
        m1_0.id=?

# 동행 게시글 생성
Hibernate: 
    insert 
    into
        accompany_post
        (accompany_area,content,created_at,custom_url,deleted_at,end_date,member_id,start_date,title,updated_at,url_qr_path) 
    values
        (?,?,?,?,?,?,?,?,?,?,?)

# 동행 게시글 상태 생성
Hibernate: 
    insert 
    into
        accompany_status
        (accompany_post_id,accompany_status_enum,created_at,deleted_at,updated_at) 
    values
        (?,?,?,?,?)

# 그룹 채팅방 생성
Hibernate: 
    insert 
    into
        chat_room
        (accompany_post_id,chat_room_type,created_at,leader_id,deleted_at,last_chat_message,last_chat_time,updated_at) 
    values
        (?,?,?,?,?,?,?,?)

# 그룹 채팅방, 동행자 추가
Hibernate: 
    insert 
    into
        chat_room_member
        (chat_room_id,created_at,deleted_at,member_id,status,updated_at) 
    values
        (?,?,?,?,?,?)
```

**Comment by ej31** [comment]
1분에 1,000명이라니.. 엄청나군요.. 🫨
